### PR TITLE
[53 log enhancements] Bonus JobStoreReaderActor and EJEA logging, batch actor timing metrics

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala [hunk 1]
```diff
@@ -20,15 +21,24 @@ class JobStoreReaderActor(database: JobStore, override val serviceRegistryActor:
...
+
     val action = database.readJobResult(head.command.jobKey, head.command.taskOutputs)
+
     action onComplete {
-      case Success(Some(result)) => head.replyTo ! JobComplete(result)
-      case Success(None) => head.replyTo ! JobNotComplete
+      case Success(Some(result)) =>
+        head.replyTo ! JobComplete(result)
+      case Success(None) =>
+        head.replyTo ! JobNotComplete
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala:32]: TOL I'd prefer not to introduce purely whitespace changes if we find ourselves reviewing this later under time constraints...

File: engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala [hunk 2]
```diff
@@ -20,15 +21,24 @@ class JobStoreReaderActor(database: JobStore, override val serviceRegistryActor:
...
       case Failure(t) =>
         log.error(t, "JobStoreReadFailure")
         head.replyTo ! JobStoreReadFailure(t)
     }
-    action.map(_ => 1)
+
+    for {
+      actionResult <- action
+      _ = log.info(s"{}: Job store check success (found something = ${actionResult.nonEmpty})", head.command.jobKey.tag)
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala:40]: ... or you could ignore my earlier comment and put this log message in the `Success` cases...

**Review by kshakir — APPROVED** [review]
+1 to minimal changes. Otherwise LGTM

File: engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala (hunk 1 above)
Comment (cjllanwarne) [reply engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala:32]: I agree. I think I originally had different logging per case before I consolidated into the for comp. Will revert these changes.

File: engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala (hunk 2 above)
Comment (cjllanwarne) [reply engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala:38]: That was my original layout. But I think this way is possibly nicer?

**Review by kshakir — APPROVED** [review]

**Review by mcovarr — APPROVED** [review]

File: engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala (hunk 2 above)
Comment (mcovarr) [reply engine/src/main/scala/cromwell/jobstore/JobStoreReaderActor.scala:38]: yeah this is fine

**Comment by cjllanwarne** [comment]
Merging to a non-develop branch despite problems in Travis because we are deliberately adding logging (which Travis cannot handle) in order to debug a production incident.

**Pull request merged**